float32 frequency
float32 amplitude
string handid